### PR TITLE
Add support for FFmpeg-Vaapi pad

### PR DESCRIPTION
**Pull request by xhaihao**

### CODE REVIEW

File: lib/ffmpeg/vaapi/vpp.py [hunk 1]
```diff
@@ -42,6 +42,14 @@ def gen_vpp_opts(self):
...
       vpfilter.append("[1:v]hwupload[v1];")
       vpfilter[-1] += "[0:v][v1]overlay_vaapi=alpha={a}".format(a = mapRange(self.alpha, [0.0, 255.0], [0.0, 1.0]))
+    elif self.vpp_op in ["pad"]:
+      vpfilter.append("pad_vaapi")
+      if hasattr(self, "padw") and hasattr(self, "padh"):
+        vpfilter[-1] += f"=width={self.padw}:height={self.padh}"
+        if hasattr(self, "x") and hasattr(self, "y"):
+          vpfilter[-1] += f":x={self.x}:y={self.y}"
+          if hasattr(self, "color"):
+            vpfilter[-1] += f":color={self.color}"
```
Comment (uartie) [inline lib/ffmpeg/vaapi/vpp.py:52]: What about `color` setting without `x` and `y`?

File: lib/ffmpeg/vaapi/vpp.py [hunk 2]
```diff
@@ -42,6 +42,14 @@ def gen_vpp_opts(self):
     elif self.vpp_op in ["overlay"]:
       vpfilter.append("[1:v]hwupload[v1];")
       vpfilter[-1] += "[0:v][v1]overlay_vaapi=alpha={a}".format(a = mapRange(self.alpha, [0.0, 255.0], [0.0, 1.0]))
+    elif self.vpp_op in ["pad"]:
+      vpfilter.append("pad_vaapi")
+      if hasattr(self, "padw") and hasattr(self, "padh"):
+        vpfilter[-1] += f"=width={self.padw}:height={self.padh}"
+        if hasattr(self, "x") and hasattr(self, "y"):
```
Comment (uartie) [inline lib/ffmpeg/vaapi/vpp.py:49]: We have to have both `x` and `y`?  Why not allow just `x` or just `y`?

File: lib/ffmpeg/vaapi/vpp.py [hunk 3]
```diff
@@ -42,6 +42,14 @@ def gen_vpp_opts(self):
     elif self.vpp_op in ["overlay"]:
       vpfilter.append("[1:v]hwupload[v1];")
       vpfilter[-1] += "[0:v][v1]overlay_vaapi=alpha={a}".format(a = mapRange(self.alpha, [0.0, 255.0], [0.0, 1.0]))
+    elif self.vpp_op in ["pad"]:
+      vpfilter.append("pad_vaapi")
+      if hasattr(self, "padw") and hasattr(self, "padh"):
+        vpfilter[-1] += f"=width={self.padw}:height={self.padh}"
```
Comment (uartie) [inline lib/ffmpeg/vaapi/vpp.py:48]: We have to have both `padw` and `padh`?  Why not allow just `padw` or just `padh`?

File: lib/ffmpeg/vaapi/vpp.py (hunk 1 above)
Comment (uartie) [inline lib/ffmpeg/vaapi/vpp.py:52]:
```suggestion
      vpopts = []
      if hasattr(self, "padw"): vpopts.append(f"width={self.padw}")
      if hasattr(self, "padh"): vpopts.append(f"height={self.padh}")
      
      # ...and so on...
      
      if len(vpopts): vpfilter[-1] += f"={':'.join(vpopts)}"
```

**Review by uartie — CHANGES_REQUESTED** [review]

File: lib/ffmpeg/vaapi/vpp.py (hunk 2 above)
Comment (xhaihao) [reply lib/ffmpeg/vaapi/vpp.py:49]: There are no dependency between `x` and `y`,  I'll follow your suggestion to update it.

File: lib/ffmpeg/vaapi/vpp.py (hunk 3 above)
Comment (xhaihao) [reply lib/ffmpeg/vaapi/vpp.py:48]: There are no dependency between `padw` and `padh`, I'll follow your suggestion to update it.

File: lib/ffmpeg/vaapi/vpp.py (hunk 1 above)
Comment (xhaihao) [reply lib/ffmpeg/vaapi/vpp.py]: Updated, thx

File: lib/ffmpeg/vaapi/vpp.py (hunk 2 above)
Comment (xhaihao) [reply lib/ffmpeg/vaapi/vpp.py]: Updated, thx

File: lib/ffmpeg/vaapi/vpp.py (hunk 3 above)
Comment (xhaihao) [reply lib/ffmpeg/vaapi/vpp.py]: Updated, thx

File: lib/ffmpeg/vaapi/vpp.py (hunk 1 above)
Comment (xhaihao) [reply lib/ffmpeg/vaapi/vpp.py]: Updated, thx

**Review by uartie — APPROVED** [review]

**Pull request merged**